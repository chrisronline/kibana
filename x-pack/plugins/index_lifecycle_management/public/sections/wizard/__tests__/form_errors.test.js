/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { ErrableFormRow } from '../form_errors';

describe('ErrableFormRow', () => {
  it('should render a form row with invalid props and mark the child as invalid', () => {
    expect(shallow(
      <ErrableFormRow
        errorKey="key"
        isShowingErrors={true}
        errors={{
          key: ['There is an error.']
        }}
      >
        <div/>
      </ErrableFormRow>
    )).toMatchSnapshot();
  });

  it('should render a form row with invalid props and mark the child as invalid only if it is invalid', () => {
    expect(shallow(
      <ErrableFormRow
        errorKey="key2"
        isShowingErrors={true}
        errors={{
          key: ['There is an error.'],
          key2: []
        }}
      >
        <div/>
      </ErrableFormRow>
    )).toMatchSnapshot();
  });

  it('should not render errors is we are not showing thenm', () => {
    expect(shallow(
      <ErrableFormRow
        errorKey="key"
        isShowingErrors={false}
        errors={{
          key: ['There is an error.'],
        }}
      >
        <div/>
      </ErrableFormRow>
    )).toMatchSnapshot();
  });
});
