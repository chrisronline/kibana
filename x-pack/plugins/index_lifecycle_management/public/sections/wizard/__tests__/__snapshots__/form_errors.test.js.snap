// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrableFormRow should not render errors is we are not showing thenm 1`] = `
<EuiFormRow
  describedByIds={Array []}
  error={
    Array [
      "There is an error.",
    ]
  }
  fullWidth={false}
  hasEmptyLabelSpace={false}
  isInvalid={false}
>
  <React.Fragment>
    <div
      isInvalid={false}
      key=".0"
    />
  </React.Fragment>
</EuiFormRow>
`;

exports[`ErrableFormRow should render a form row with invalid props and mark the child as invalid 1`] = `
<EuiFormRow
  describedByIds={Array []}
  error={
    Array [
      "There is an error.",
    ]
  }
  fullWidth={false}
  hasEmptyLabelSpace={false}
  isInvalid={true}
>
  <React.Fragment>
    <div
      isInvalid={true}
      key=".0"
    />
  </React.Fragment>
</EuiFormRow>
`;

exports[`ErrableFormRow should render a form row with invalid props and mark the child as invalid only if it is invalid 1`] = `
<EuiFormRow
  describedByIds={Array []}
  error={Array []}
  fullWidth={false}
  hasEmptyLabelSpace={false}
  isInvalid={false}
>
  <React.Fragment>
    <div
      isInvalid={false}
      key=".0"
    />
  </React.Fragment>
</EuiFormRow>
`;
